components/profile/components/EditProfile/index.js

```diff
@@ -0,0 +1,151 @@
+                        <Input />
+                    </div>
+                    <div className={`${infoField}`}>
+                        <h4>{lastNameLabel}</h4>
+                        <Input />
+                    </div>
+                    <div className={`${infoField}`}>
+                        <h4>{profilePhotoLabel}</h4>
+                        <Upload
+                            listType={EDIT_PROFILE_UPLOAD_LIST_TYPE}
+                            fileList={fileList}
+                            beforeUpload={() => false}
+                            onPreview={this.handleImagePreview}
+                            onChange={this.handleImageChange}
+                        >
+                            {fileList.length < 1 ? (
+                                <div>
+                                    <Icon type={plus} />
+                                    <div className={antUploadText}>{uploadLabel}</div>
+                                </div>
+                            ) : null}
+                        </Upload>
+                        <Modal
+                            visible={previewVisible}
+                            footer={null}
+                            onCancel={this.handleImagePreviewCancel}
+                        >
+                            <img
+                                alt={profilePhotoPreviewAlt}
+                                className={`${w100}`}
+                                src={previewImage}
+                            />
+                        </Modal>
+                    </div>
+
+                </Col>
+                <Col span={12} md={12} xs={24}>
+                    <div className={`${infoField}`}>
+                        <h4>{highSchoolLabel}</h4>
+                        <Input />
+                    </div>
+                    <div className={`${infoField}`}>
+                        <h4>{intendedUniLabel}</h4>
+                        <Input />
+                    </div>
+                    <div className={`${infoField}`}>
+                        <h4>{locationLabel}</h4>
+                        <Input value={location} className={`${locationInput}`}  />
+                        <Button 
+                            onClick={this.getLocation} 
+                            type={BUTTON_PRIMARY} 
+                            loading={locationLoading}
+                        >
+                            {getLocationLabel}
+                        </Button>
+                    </div>
+                </Col>
+                <Col span={24} className={`${mb2}`}>
+                    <div className={`${infoField}`}>
+                        <h4>{bioLabel}</h4>
+                        <TextArea className={`${w80}`} rows={4} />
+                    </div>
+                </Col>
+                <Col span={24}>
+                    <Button type={BUTTON_PRIMARY}>{EDIT_PROFILE_TITLE}</Button>
+                </Col>
+            </Row>
+        );
+    }
+}
```

Comment:
This file has alot of repeated code, can you refactor it by mapping over an array of objects to create these code chunks. This is very messy as it stands, please also use a similar pattern in other parts of your code base

Exampl:
```
[{}, {}, {}].map(item => {
    const { className, label, data } = item;
    // use data to determine that
    const element = generateElement(data); // you also don't have to call it element
    
    return (
        <div className={`${infoField}`}>
            <h4>{lastNameLabel}</h4>
            { element }
        </div>
    )
});
```

Reply:
I thought so too. That generateElement is a great idea. Thanks